tools/autograd/derivatives.yaml

```diff
@@ -160,16 +175,25 @@
 # in Decalarations.yaml
 - name: abs(Tensor self) -> Tensor
   self: grad * self.sgn()
+  result: auto_element_wise
 
 - name: acos(Tensor self) -> Tensor
   self: grad * -((-self * self + 1).rsqrt()).conj()
+  result: auto_element_wise
 
 - name: add.Tensor(Tensor self, Tensor other, *, Scalar alpha=1) -> Tensor
   self: handle_r_to_c(self.scalar_type(), grad)
   other: handle_r_to_c(other.scalar_type(), maybe_multiply(grad, alpha.conj()))
+  result: "self_fw_grad.defined()? (other_fw_grad.defined()? maybe_multiply(self_fw_grad, alpha) + other_fw_grad : maybe_multiply(self_fw_grad, alpha)) : (other_fw_grad.defined()? other_fw_grad : Tensor())"
```

Comment:
This looks wrong. `add(self, other, alpha) == self + other * alpha`. Then, `self + dself + (other + dother) * alpha` = `self + other * alpha + dself + dother * alpha`. alpha scaling should apply to other, not self.

Reply:
I'm curious to know why this wasn't caught by testing in the subsequent PR, it seems like we should have an existing gradcheck for non-one scalar on add.

Reply:
Comparison with JAX:

```
from jax import jvp
import jax.numpy as jnp

def add(self, other, alpha=2):
  return self + other * alpha

print(jvp(add, (jnp.zeros(3), jnp.ones(3)), (jnp.zeros(3), jnp.ones(3))))
```

Reply:
All the other add definitions look sus too